### PR TITLE
Refactor entrypoint to support pushes and be faster on PRs

### PR DESCRIPTION
**Pull request by underyx**
Closes https://github.com/returntocorp/bento/issues/298 and closes https://github.com/returntocorp/bento/issues/291 as well

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -5,31 +5,95 @@ shopt -s nullglob globstar
...
+  echo "== [ERROR] this action only works on pull_request or push events"
+  echo
+  echo ".github/workflows/bento.yml"
+  echo "---------------------------"
+  echo
+  echo "name: Bento"
+  echo "on: [pull_request]"
+  echo "[...]"
+  exit 2
+}
```
Comment (dlukeomalley) [inline entrypoint.sh]: What do you think about something more detailed, like:

> the `on:` key of `.github/workflows/bento.yml` contains an unsupported value. Please use one of the following values: `pull_request` or `push`. If the problem persists, please file an issue on https://github.com/returntocorp/bento/issues/new/choose"

Comment (underyx) [reply entrypoint.sh]: > the `on:` key of `.github/workflows/bento.yml` contains an unsupported value.

Just wanna make sure, is it okay to make the assumption that this is the path to the config file? We don't know the actual file path, and people could be using a different one.

>  Please use one of the following values: `pull_request` or `push`. 

Well, now it reads like an automatic parser-generated message, which implies that the message is gonna be technically correct and exhaustive. But other than the above inaccuracy with the file path, this assumes there's one "value" but the `on:` key can contain a list of events, such as `on: [push, pull_request]`, or an object with more config options. That's why I wanted to word the error message more to explain what the actual problem is for those who understand how GH Actions works, and then provide the simplest fix for people who don't want to bother understanding.

Imagine if someone uses this configuration:

```yaml
on:
  push:
  pull_request:
  schedule:
    - cron: '0 * * * *'
```

We'd fail on their scheduled run and there "Please use one of the following values: `pull_request` or `push`" is incorrect advice. We could instead write "Please use only the following values: `pull_request` or `push`." which is a bit better? But I'm still not sure if explaining the configuration file's options, which could have more complexities (e.g. what if GitHub starts allowing composition of the config file via importing smaller files) is preferable to explaining the problem with the job's state: that it was triggered for the wrong event for whatever reason.

> If the problem persists, please file an issue on https://github.com/returntocorp/bento/issues/new/choose

I think this error is always going to be a config issue, so maybe we could suggest the support email instead? It seemed like that's what we normally do for config issues in bento-core.

Comment (dlukeomalley) [reply entrypoint.sh]: I agree. Overall I do think it is clearer and more actionable though, perhaps we can incorporate your ideas? I worry that GitHub Actions are not widely adopted yet and using specialized terminology like "event" won't be understood by common users (I had to do reading myself on what events were).

In response to your feedback: 

> the Bento Check action was triggered by an unsupported GitHub event. Often this is caused by an unsupported value for `on` in the actions configuration. To resolve this issue, please confirm that the `on` key only contains values from the following list: [pull_request, push]. If the problem persists, please file an issue on https://github.com/returntocorp/bento/issues/new/choose"

**Comment by nbrahms** [comment]
Closes https://github.com/returntocorp/bento-core/issues/754

File: entrypoint.sh (hunk 1 above)
Comment (underyx) [reply entrypoint.sh]: > My preference is always for public tickets vs. private emails

Glad to hear! I was worried on cause of the precedent in bento-core.

I like this new version! Committed & merging.

**Pull request merged**